src/rgw/driver/rados/rgw_data_sync.cc

```diff
@@ -4916,6 +4928,8 @@ class RGWBucketShardIncrementalSyncCR : public RGWCoroutine {
 int RGWBucketShardIncrementalSyncCR::operate(const DoutPrefixProvider *dpp)
 {
   int ret;
+  jspan trace;
+  jspan_context trace_ctx{false, false};
```

Comment:
it looks like `trace_ctx` is unused

for `trace`, this function is a stackless coroutine, meaning that the stack goes away each time it suspends. i think we should only construct `trace` in the scope just before `spawn(new SyncCR(...`, and move it into that class. that would extend the span's lifetime until the object's sync completes